fms/models/__init__.py

```diff
@@ -208,8 +258,8 @@ def _is_dp(distributed_strategy):
 
 
 def get_model(
-    architecture: str,
-    variant: str,
+    architecture: Optional[str] = None,
```

Comment:
i wonder if it might make sense to call this something explicit like `architecture: str = 'infer'` or `= hf_model_or_path` or something like that, instead of having it be an optional.

Reply:
also, if someone says something like `model_path="meta/llama-2-7b"` (or whatever the key is), how are we differentiating if we should be loading the weights (i.e. calling `AutoModel.from_pretrained`) vs just loading the model from config (in which case we may want to train it ourselves)?

It looks like what we have doesn't yet support both use-cases, but we should consider what the API will look like when it does. For the existing code, the optionality of the model_path parameter is what determines if we're loading a randomly initialized vs pre-trained model, so it would be ideal if there was some sensible way to make this conform at least somewhat to that usage.

Reply:
We never call `AutoModel.from_pretrained`. This does bring up a good question though, should `get_model` handle the creation of the uninitialized model here. We could just tell a user to explicitly call `reset_parameters`? Or we can just make the assumption that if the weights exist in the path, load them, otherwise it will be initialized to train

Reply:
in the case where the key is remote, calling reset_parameters after the fact could mean first downloading a lot of unnecessary weight data. it would be preferable if the api supports that directly (even if we don't actually implement that path yet now)

Reply:
fixed in latest commit

Reply:
We are now going to have special strings for architecture "hf_pretrained", "hf_configured" which will then allow use of variant for `model_id_or_path`. If using hf_pretrained, the weights will be downloaded and loaded. If using hf_configured, just the model configuration will be created, and we will follow normal model_path and source logic.